tee-worker/litentry/primitives/src/identity.rs

```diff
@@ -33,10 +35,60 @@ use sp_std::vec::Vec;
 use strum_macros::EnumIter;
 
 pub type MaxStringLength = ConstU32<64>;
-pub type IdentityString = BoundedVec<u8, MaxStringLength>;
+pub type IdentityInnerString = BoundedVec<u8, MaxStringLength>;
+
 pub type HashedAddressMapping = GenericHashedAddressMapping<BlakeTwo256>;
 
-#[derive(Encode, Decode, Copy, Clone, Debug, Default, PartialEq, Eq, TypeInfo, MaxEncodedLen)]
+impl Decode for IdentityString {
+	fn decode<I: Input>(input: &mut I) -> Result<Self, Error> {
+		let inner: BoundedVec<u8, MaxStringLength> = Decode::decode(input)?;
+		Ok(IdentityString { inner })
+	}
+}
+
+impl Encode for IdentityString {
+	fn encode(&self) -> Vec<u8> {
+		let mut res = Vec::new();
+		self.inner.encode_to(&mut res);
+		res
+	}
+}
+
+#[derive(Eq, PartialEq, Clone, MaxEncodedLen, Default)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub struct IdentityString {
```

Comment:
Do I understand it correctly that we need such a flattened struct because of prod/non-prod `Debug` implementation?

Reply:
Yes, the actual reason is just to not introduce any breaking changes and to not pollute api contract with this implementation detail. 
This struct will be serialized/deserialized the same as it was before so for the consumers there would be no change.